### ISSUE TITLE
Cannot create a TensorRT session with a Debug build

### ISSUE THREAD

**Issue by JulienTheron** [microsoft/onnxruntime#16273]
### Describe the issue

Having built the ORT 1.15.0 in Debug, I cannot create a session with TensorRT options.
CreateSession throws a std::bad_array_new_length exception. It runs fine when the library is built in Release.

See log below:
`2023-06-07 14:19:13.0961372 [E:onnxruntime:, inference_session.cc:1645 onnxruntime::InferenceSession::Initialize::<lambda_afad4075fd15ce1a09f4907d66e057d6>::operator ()] Exception during initialization: bad array new length
Exception during initialization: bad array new length`

### To reproduce

1. Build the library in Debug configuration. This is what I used:
`build.bat --config Debug --build_shared_lib --parallel --skip_tests --disable_memleak_checker --use_tensorrt --cuda_home "%CUDA_PATH_V11_8%" --cuda_version 11.8 --cudnn_home "%CUDNN_PATH_V8_9_0%" --tensorrt_home "%TENSORRT_PATH_V8_6_1%"`
2. Create a session with TensorRT options. I used the FNS Candy sample.
```
const OrtApi* g_ort = OrtGetApiBase()->GetApi(ORT_API_VERSION);

OrtEnv* env;
ORT_ABORT_ON_ERROR(g_ort->CreateEnv(ORT_LOGGING_LEVEL_WARNING, "test", &env));

OrtSessionOptions* session_options;
ORT_ABORT_ON_ERROR(g_ort->CreateSessionOptions(&session_options));

OrtTensorRTProviderOptions o;
memset(&o, 0, sizeof(o));
OrtStatus* onnx_status = g_ort->SessionOptionsAppendExecutionProvider_TensorRT(session_options, &o);

OrtSession* session;
ORT_ABORT_ON_ERROR(g_ort->CreateSession(env, model_path, session_options, &session));
```

### Urgency

As this issue only seems to affect the Debug build, it is not urgent but it's definitely concerning that it occurs in the first place.

### Platform

Windows

### OS Version

22621.1702

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

v1.15.0

### ONNX Runtime API

C

### Architecture

X64

### Execution Provider

TensorRT

### Execution Provider Library Version

TensorRT 8.6.1 CUDA 11.8 cuDNN 8.9.0

**Comment by simonjub** [comment]
I'm seeing the same issue in debug, no matter which model I use to create a session. onnxruntime 1.15.0, TensorRT 8.6.1.6, cuDNN 8.9.2.26, CUDA 11.8.0

When debugging, I tracked the error to this line but I cannot step into this call. It throws an exception but I can't tell what's wrong with the parameters if anything: https://github.com/microsoft/onnxruntime/blob/b07b647f66b1b02514aae41e0710a698b73fed9e/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L1419

**Comment by yf711** [comment]
Thanks for pointing this out. 
I've repro this issue locally with either the debug build of 1.15.0 or latest main branch, and I will continue checking on this.

**Comment by AndreyOrb** [comment]
Having the same issue and can confirm that it comes from the line:
`trt_parser->supportsModel(string_buf.data(), string_buf.size(), parser_nodes_list, model_path_); `

**Comment by AndreyOrb** [comment]
The problem persists for both TRT 8.6.1.6 and TRT 8.6.0.12

**Comment by chilo-ms** [comment]
This issue only happens when ORT TRT links against TRT built-in parser (The default setting when building ORT TRT).
If you build ORT TRT with `--use_tensorrt_oss_parser` meaning ORT TRT uses [open source TRT parser](https://github.com/onnx/onnx-tensorrt), it won't have this issue.
We will raise this issue with Nvidia and get back to you soom.

**Comment by AndreyOrb** [comment]
Is there any news from Nvidia?

**Comment by JulienTheron** [comment]
Fixed in 1.16.0.
